### ISSUE TITLE
does python-iview still work?

### ISSUE THREAD

**Issue by jonathon-love** [sramsay64/python-iview#10]
hi,

first, you have done an absolutely fabulous job maintaining and developing python-iview. i (and many others) thoroughly appreciate it.

but it looks to me that newer programs are not downloading correctly. it started with season 3 of adventure time. season 2 i could download and view fine, but downloading season 3, VLC et al. would not play it.

it also looks like other programs are affected.

has this been your experience?

with thanks

**Comment by sramsay64** [comment]
I've also noticed this. I'm rather busy at the moment but I might get time to look at the bug in a few days time.

**Comment by lachlan-00** [comment]
I've just noticed this as well. All new files are different/incomplete. ffmpeg can't convert them to anything so the downloads are bad.

I redownloaded the same files and the new file is larger than the working copies which i overwrote before realising.

**Comment by sramsay64** [comment]
Yes. Rather sadly python-iView no longer works. I've had a bit of a look into the internals of it and I don't understand how it works well enough to know what's going wrong. Although my guess at what the problem is This: python-iView is corrupting the metadata of the flv file by passing some data that came directly from the internet. It's programmed to expect a specific bit of data from iView and iView has changed what it returns in that case (it returns something extra for example). That data is then passed straight in to the flv file which corrupts it.

Not sure if I'm right about that but in any case I'd love to solve it but I'm still not sure what the problem is after several hours work on it and I don't have the time/knowledge to do this. I'd absolutely love it if one of the original guys who had the project before me could help. They may not want to for legal reasons though.

So sadly for now the project is dead.

**Comment by nemothorx** [comment]
For my usage, it's still downloading fine - files playable in kodi and mpv without issue. No hints of flv corruption that I've seen.

(The glitch I *am* seeing however is that the program listings gets an old version from several days ago, and so I don't see new shows. (I save the program list several times a day). It's unclear if this is a glitch with python-iview, or with the iview site (I see new shows on the webUI, but I know python-iview emulates some specific device)

I'm sad to see this considered a dead project though, since it still works for me. I'm not in a position to help code, but happy to provide details of software versions I have, and resultant downloaded files, if that will help any.

**Comment by sramsay64** [comment]
@earthnative What!! It's working, that's awesome :). The project isn't dead after all. Thanks so much for telling me this :). However it seems to only be `iview-gtk` that's working. But I'll be able to fix all of the rest fairly simply. But thanks a million It all seems to be working just fine now. (There defiantly was a period where files were being downloaded but corruptly (as 3 of us reported) though).

**Comment by jonathon-love** [comment]
Hahahaha, this is awesome. thanks guys!

**Comment by nemothorx** [comment]
It's curious that my experience is so different... my iview has never failed (I just checked over 100 videos downloaded between October and May and found none that fail to play or even show hint of corruption). I note that I only use iview-cli (and some home-made wrapper scripts to cache the listings, and report on what's changed). I've never use iview-gtk normally, but testing now and it failed first time, but worked second - with an up-to-date episode listing. iview-ng failed multiple times (timeout) until I curl'd the listing and then it worked (either coincidence, or I triggered a proxy cache finally?)

All very curious, but for my usage, the episode listings via the the original cli version would be awesome! (The important thing for me is ability to download direct with a single command :)

**Comment by lachlan-00** [comment]
I just checked a few series and one file worked and the rest didn't

**Comment by nemothorx** [comment]
listing downloadings are working again as of this morning. No code changes or anything, just came good from whatever was glitching upstream. 

\o/

**Comment by sramsay64** [comment]
Just thought I might point out that https://youtube-dl.org/ seems to be able to download from iView. It sometimes has issues and a can corrupt few seconds of the video but it's currently a decent replacement for python-iView. Which is at this point dead until someone picks it up again.

So for those of you following this issue wanting to download things from iView, [youtube-dl](https://youtube-dl.org/) is a replacement that works now.

**Comment by nemothorx** [comment]
Thank you, that's good to know!

(I started getting unplayable files in python-iview a few months ago, but life has been too busy to worry about it. I've still been downloading the resultant files in the hope that a future video player update may play them, or I'd find a way to convert them to something playable!)

**Comment by lachlan-00** [comment]
yt-dl does work it may depend on your version but i've tested a few videos.

The broken files you've downloaded will probably never work. I think it's downloading extra things at the start/end of the download corrupting the file.

**Comment by nemothorx** [comment]
My casual and naive inspection of the files had me thinking it was missing stuff at the start of the file - which is worse, since extra cruft is theoretically easier to remove... 

It's no biggie. Anything I get with yt-dl I can remove from the broken collection, and see how much I care about what's left, later on...

(yt-dl doco hints it can do SBS too. That's exciting)
